src/EFCore.Relational/Migrations/Internal/MigrationExtensions.cs

```
@@ -21,6 +23,6 @@ public static class MigrationExtensions
         ///     doing so can result in application failures when updating to a new Entity Framework Core release.
         /// </summary>
         public static string GetId([NotNull] this Migration migration)
-            => migration.GetType().GetCustomAttribute<MigrationAttribute>()?.Id;
+            => migration.GetType().GetCustomAttribute<MigrationAttribute>()!.Id;
```

Should we throw a more informative exception in case the attribute is missing? Or are we sure that this method only gets called for types which have it?

---

This is only ever used on migrations found by looking for types that have the attribute.